terraform/persistent/resources/main.tf

```diff
@@ -0,0 +1,105 @@
+# SPDX-FileCopyrightText: 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+
+provider "azurerm" {
+  # https://github.com/hashicorp/terraform-provider-azurerm/issues/24804
+  skip_provider_registration = true
+  features {}
+}
+
+terraform {
+  required_providers {
+    azurerm = {
+      source = "hashicorp/azurerm"
+    }
+    secret = {
+      source = "numtide/secret"
+    }
+  }
+}
+
+################################################################################
+
+terraform {
+  # Backend for storing terraform state (see ../../state-storage)
+  backend "azurerm" {
+    # resource_group_name and storage_account_name are set by the callee
+    # from command line in terraform init, see terraform-init.sh
+    container_name = "ghaf-infra-tfstate-container"
+    key            = "ghaf-infra-persistent.tfstate"
+  }
+}
+
+################################################################################
+
+# Variables
+variable "persistent_resource_group" {
+  type        = string
+  description = "Parent resource group name"
+}
+
+locals {
+  # Raise an error if workspace is 'default',
+  # this is a workaround to missing asserts in terraform:
+  assert_workspace_not_default = regex(
+    (terraform.workspace == "default") ?
+  "((Force invalid regex pattern)\n\nERROR: workspace 'default' is not allowed" : "", "")
+
+  # Sanitize workspace name:
+  ws = substr(replace(lower(terraform.workspace), "/[^a-z0-9]/", ""), 0, 16)
+}
+
+# Data source to access persistent resource group (see ../main.tf)
+data "azurerm_resource_group" "persistent" {
+  name = var.persistent_resource_group
+}
+
+# Current signed-in user
+data "azurerm_client_config" "current" {}
+
+
+################################################################################
+
+# Resources
+
+# secret_resouce must be created on import, e.g.:
```

Comment:
should be "secret_resource"

Reply:
Fixed in https://github.com/tiiuae/ghaf-infra/commit/eb4eb5ae550bcd4ab71ac97ea1733ba79bb3490f